packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -135,15 +140,19 @@ function readTreeSection(node: NodeCore): {
 					// "name": <node name>
 					// "children": <blob id>
 					if (content === "children") {
-						const result = readTreeSection(treeNode.getNode(3));
+						const result = readTreeSection(treeNode.getNode(3), blobContents);
 						trees[treeNode.getString(1)] = result.snapshotTree;
 						slowTreeStructureCount += result.slowTreeStructureCount;
 						continue;
 					}
 					// "name": <node name>
 					// "value": <blob id>
 					if (content === "value") {
-						snapshotTree.blobs[treeNode.getString(1)] = treeNode.getString(3);
+						const blobId = treeNode.getString(3);
+						snapshotTree.blobs[treeNode.getString(1)] = blobId;
+						if (!blobContents.has(blobId)) {
+							snapshotTree.omitted = true;
```

Comment:
This feels like required, but not sufficient. I.e. we want to propagate this flag up the tree to every single node, including root of the tree. It actually becomes not "omitted", but more "this sub-tree has missing blobs.

If we do not propagate it up the stack, then only the leaf sub-tree in DDS summary will have it - data store node will not have it, right?

Reply:
I think actually the best course of action (of possible) - not do it in the driver, but scan the tree in data store load path.
That assumes that we have this info at container runtime layer. I think we should, as we rely on a new snapshot format, that includes all the blobs, no?

Reply:
No, we are adding it everywhere on each subtree.

Reply:
While parsing each tree, if any blobid is missing in contents, then we are marking that node as omitted on snapshotTree.

Reply:
I can see that if we do it in container runtime then we need to do it once for all drivers but in that case, we need to go over the trees structure again to set this flag where in driver, we don't have to bear any extra cost although it should be small cost to bear. And we have to do it during partial snapshot fetches too and set omitted flag in those cases too.